### DISCUSSION TITLE
Using `deno compile` to create distributable Fresh app

### DISCUSSION THREAD

**Discussion by cngarrison** [freshframework/fresh#2639] [Q&A]
Is it possible to create compiled binary for Deno Fresh app? 

I've tried various `deno compile` options and can get a binary built, but it won't run without errors.

When compiling `src/main.ts`, running the binary throws:

> Error: Could not find a deno.json file in the current directory or any parent directory.

When compiling `src/_fresh/main.js`, running the binary throws:

> ReferenceError: history is not defined

Is there a workaround or solution for compiling a binary I can distribute so users can run the Deno Fresh app locally?

**Comment by marvinhagemeister** [comment]
Not possible at the moment, because `deno compile` cannot track non TS/JS assets.

**Reply by cngarrison** [reply]
Thanks for clarifying where the issue is. 

> cannot track non TS/JS assets.

Is that referring to JSON files? Specifically `deno.json`? 

If that is what you are referring to, can I get Fresh to inline the JSON? I'd be happy to work on a solution for that.

**Reply by marvinhagemeister** [reply]
You could find a solution for this particular problem, but the next one you'll be running into is all the static assets like CSS files, images, etc. The way `deno compile` works is that it follows the import statements and only includes files that are reachable through that. Since the static files are not directly referenced in an import statement, they will not be included.

**Reply by cngarrison** [reply]
Hehe, I was anticipating that. :-)

The app doesn't have many assets. There are no images, and a single CSS file for Tailwind plugin. I figured if I worked out how to inline (& use) the JSON, I could do the same for the CSS. 

I had already tried importing `deno.json` in `main.ts`:

`import * as denoConfig from './deno.json' assert { type: "json" };`

But I believe Fresh is doing it's own loading of the file and that import didn't help. 

Do you happen to know the status of being able to track non JS/TS assets in the Deno project? IOW, would be worthwhile for me to submit an issue? (I'll search for it if you're not sure.)
